localstack-core/localstack/services/stepfunctions/asl/eval/event/execution_event_manager.py

```diff
@@ -0,0 +1,209 @@
+        event_id: int,
+        source_event_id: int,
+        event_type: HistoryEventType,
+        timestamp: datetime.datetime,
+        event_details: Optional[EventDetails],
+    ):
+        history_event = self._create_history_event(
+            event_id=event_id,
+            source_event_id=source_event_id,
+            event_type=event_type,
+            timestamp=timestamp,
+            event_details=event_details,
+        )
+        self._history_event_list.append(history_event)
+
+    @staticmethod
+    def _remove_data_from_history_log(details_body: dict) -> None:
+        remove_keys = ["input", "inputDetails", "output", "outputDetails"]
+        for remove_key in remove_keys:
+            details_body.pop(remove_key, None)
+
+    @staticmethod
+    def _create_history_log(
+        event_id: int,
+        source_event_id: int,
+        event_type: HistoryEventType,
+        timestamp: datetime.datetime,
+        execution_arn: LongArn,
+        event_details: Optional[EventDetails],
+        include_execution_data: bool,
+    ) -> HistoryLog:
+        log = HistoryLog(
+            id=str(event_id),
+            previous_event_id=str(source_event_id),
+            event_timestamp=timestamp,
+            type=event_type,
+            execution_arn=execution_arn,
+        )
+        if event_details:
+            if len(event_details) > 1:
+                LOG.warning(f"Event details with multiple bindings: {to_json_str(event_details)}")
+            details_body = next(iter(event_details.values()))
+            if not include_execution_data:
+                # clone the object before modifying it as the change is limited to the history log value.
+                details_body = copy.deepcopy(details_body)
+                ExecutionEventManager._remove_data_from_history_log(details_body=details_body)
+            log["details"] = details_body
+        return log
+
+    def _publish_history_log(
+        self,
+        event_id: int,
+        source_event_id: int,
+        event_type: HistoryEventType,
+        timestamp: datetime.datetime,
+        event_details: Optional[EventDetails],
+    ):
+        # No logging session for this execution.
+        if self._cloud_watch_logging_session is None:
+            return
+
+        # This event is not recorded by this execution's logging configuration.
+        if not self._cloud_watch_logging_session.log_level_filter(history_event_type=event_type):
+            return
+
+        history_log = self._create_history_log(
+            event_id=event_id,
+            source_event_id=source_event_id,
+            event_type=event_type,
+            timestamp=timestamp,
+            execution_arn=self._cloud_watch_logging_session.execution_arn,
+            event_details=event_details,
+            include_execution_data=self._cloud_watch_logging_session.configuration.include_execution_data,
+        )
+        self._cloud_watch_logging_session.publish_history_log(history_log=history_log)
```

Comment:
Could it make sense to add a LocalStack logger here as well for easier visibility if debug logs are enabled?